src/disco/shred/fd_shredder.c

```diff
@@ -82,6 +82,31 @@ fd_shredder_delete( void *          mem      ) {
 }
 
 
+fd_shredder_t *
+fd_shredder_skip_batch( fd_shredder_t * shredder,
+                        ulong           entry_batch_sz,
+                        ulong           slot ) {
+
+  if( FD_UNLIKELY( entry_batch_sz==0UL ) ) return NULL;
+
+  if( FD_UNLIKELY( slot != shredder->slot ) ) {
+    shredder->data_idx_offset   = 0UL;
+    shredder->parity_idx_offset = 0UL;
+  }
+
+  ulong entry_bytes           =  entry_batch_sz - shredder->offset;
+  ulong chunk_size            =  fd_ulong_if( entry_bytes>=2UL*31200UL, 31200UL, entry_bytes );
```

Comment:
I don't think this is right. You should get rid of `chunk_size`, since in this case it's the same as `entry_batch_sz`.